reconstruct/reconstruct.go

```diff
@@ -101,9 +100,72 @@ func ReconAccuracy(simFilename string, reconFilename string, leavesOnlyFile stri
 	acc := 100 - accuracy
 	answer[allNodes] = acc
 
+	return answer, accByBase
+}
+
+// ReconAccuracyByBase will run if the option given to ReconAccuracy = 1. This function calculates the percentage of first, second and third bases in codons that were correct.
+func ReconAccuracyByBase(simFilename string, reconFilename string, gpFilename string) map[string][]float64 {
+	sim := fasta.Read(simFilename)
+	rec := fasta.Read(reconFilename)
+	recon := fasta.ToMap(rec)
+	genes := genePred.Read(gpFilename)
+	answer := make(map[string][]float64)
+
+	for s := 0; s < len(sim); s++ {
+		var percentage0, percentage1, percentage2 float64
+		var mistakes0, mistakes1, mistakes2 float64
+		var total0, total1, total2 float64
+		rSeq, ok := recon[sim[s].Name]
+		if ok {
+			for i := 0; i < len(sim[s].Seq); i++ {
+				for g := 0; g < len(genes); g++ {
+					inExon, exon := simulate.CheckExon(genes[g], i)
+					if inExon {
+						loc := calcLocationInCodon(genes[g], exon, i)
+						switch loc {
+						case 0:
+							total0 += 1
+							if sim[s].Seq[i] != rSeq[i] {
+								mistakes0 += 1
+							}
+						case 1:
+							total1 += 1
+							if sim[s].Seq[i] != rSeq[i] {
+								mistakes1 += 1
+							}
+						case 2:
+							total2 += 1
+							if sim[s].Seq[i] != rSeq[i] {
+								mistakes2 += 1
+							}
+						}
+					}
+				}
+			}
+			percentage0 = (mistakes0 / total0) * 100
+			percentage1 = (mistakes1 / total1) * 100
+			percentage2 = (mistakes2 / total2) * 100
+			answer[sim[s].Name] = append(answer[sim[s].Name], 100-percentage0)
+			answer[sim[s].Name] = append(answer[sim[s].Name], 100-percentage1)
+			answer[sim[s].Name] = append(answer[sim[s].Name], 100-percentage2)
+		}
+	}
 	return answer
 }
 
+// findLocationInCodon returns what position in a codon a given base inhabits (0, 1, 2)
+func calcLocationInCodon(gene genePred.GenePred, exon int, position int) int {
+	var positionInCodon int
+
+	location := position - gene.ExonStarts[exon] + gene.ExonFrames[exon]
+	if location%3 != 0 {
```

Comment:
Do we need this "if/else"?  It seems like going through this "if" is the same as just having the line:
positionInCodon = location % 3